src/views/CoCurricularTranscript/Components/CoCurricularTranscriptActivity/index.js

```diff
@@ -1,80 +1,132 @@
-        </Grid>
-        <div className="divider">
-          <Divider />
-        </div>
-        <Grid />
-      </div>
-    );
-    if (!this.props.isUnique) {
-      return (
-        <div className="divider">
-          <Divider light={true} />
-        </div>
-      );
+  // Helper functions for parsing and translating sessionCode which is of the format "YYYYSE"
+  // where SE is 09 for fall, 01 for spring, 05 for summer
+
+  // Returns: string of year in format YYYY
+  sliceYear = sesCode => {
+    return sesCode.toString().slice(0, 4);
+  };
+
+  // Returns: string of month (Mon), month being the first month of the given semester
+  sliceStart = sesCode => {
+    switch (sesCode.toString().slice(4, 6)) {
+      case '09':
+        return 'Sep';
+      case '01':
+        return 'Jan';
+      case '05':
+        return 'May';
+      default:
+        console.log('An unrecognized semester code was provided');
+        return '';
     }
-    return heading;
   };
 
-  getContent = () => {
-    const { Activity } = this.props;
+  // Returns: string of month (Mon), month being last month of the given semester
+  sliceEnd = sesCode => {
+    switch (sesCode.toString().slice(4, 6)) {
+      case '09':
+        return 'Dec';
+      case '01':
+        return 'May';
+      case '05':
+        return 'Sep';
+      default:
+        console.log('An unrecognized semester code was provided');
+        return '';
+    }
+  };
 
-    let content = (
-      <Grid container>
-        <Grid item xs={6}>
-          <List>
-            <Typography className="text"> {Activity.ActivityDescription} </Typography>
-          </List>
-        </Grid>
-        <Grid item xs={6}>
-          <List>
-            <Typography className="text"> {Activity.ParticipationDescription} </Typography>
-          </List>
-        </Grid>
-      )
-    return content;
-};*/
+  // Param: expects an array of [month in which the earlier session ended,
+  //                             year in which the ealier session ended,
+  //                             month in which the later session started (format: Mon),
+  //                             year in which the later session started (YYYY)]
+  // Returns: true if given month-year pairs are consecutive, false otherwise. Summers are not
+  //        considered a break consecutiveness because there are no summer activities. */
+  checkConsecutiveness = dates => {
+    return (
+      dates[1] === dates[3] ||
+      (parseInt(dates[1], 10) + 1 === parseInt(dates[3], 10) &&
+        dates[0] === 'Dec' &&
+        dates[2] === 'Jan')
+    );
+  };
+
+  // Prepares a list of sessions to be displayed as one coherent string representing the timepsan
+  // of the membership.
+
+  // Param: sessionsList - a list of sessionCodes
+  // Returns: A string representing the duration of the user's membership based on the sessionsList
+  formatDuration = sessionsList => {
+    let duration = '';
+    sessionsList.sort(function(sessA, sessB) {
```

Comment:
Would it be better to be consistent and use react function syntax?

Reply:
An arrow function wouldn't work here, since the function is being defined as the parameter of the sort function.